src/Microsoft.Azure.SignalR/HubHost/ServerLifetimeManager.cs

```diff
@@ -0,0 +1,42 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Hosting;
+
+namespace Microsoft.Azure.SignalR.HubHost
+{
+    class ServerLifetimeManager
+    {
+        private readonly IList<Func<Task>> _shutdownHooks = new List<Func<Task>>();
+
+        public ServerLifetimeManager(
+            IServiceProvider provider
+        )
+        {
+#if NETCOREAPP
+            var lifetime = provider.GetService<IHostApplicationLifetime>();
+#elif NETSTANDARD
+            var lifetime = provider.GetService<IApplicationLifetime>();
+#else

```

Comment:
`else throw` for safety

Reply:
I think that disable the feature is enough in this case.
Not very necessary to throw an exception if our SDK ran in an unsupported framework.

Reply:
it is the build time throw not runtime throw